### ISSUE TITLE
Links to hydra builds of platforms other than x86_64-linux and aarch64-linux are dead

### ISSUE THREAD

**Issue by iFreilicht** [NixOS/nixos-search#748]
When searching for a package (let's take `moreutils`), links to each pacakges hydra runs are shown in the details view:

![Screenshot 2024-03-28 at 15 50 09](https://github.com/NixOS/nixos-search/assets/9742635/69e3f37e-3834-4d80-b688-17317258da7a)

> - [aarch64-darwin](https://hydra.nixos.org/job/nixos/release-23.11/nixpkgs.moreutils.aarch64-darwin)
> - [x86_64-linux](https://hydra.nixos.org/job/nixos/release-23.11/nixpkgs.moreutils.x86_64-linux)
> - [i686-linux](https://hydra.nixos.org/job/nixos/release-23.11/nixpkgs.moreutils.i686-linux)
> - [aarch64-linux](https://hydra.nixos.org/job/nixos/release-23.11/nixpkgs.moreutils.aarch64-linux)
> - [x86_64-darwin](https://hydra.nixos.org/job/nixos/release-23.11/nixpkgs.moreutils.x86_64-darwin)

These links are valid for `x86_64-linux` and `aarch64-linux`, but not for any other platforms, because the prefix is hardcoded to `https://hydra.nixos.org/job/nixos/release-23.11/` (or `nixos/trunk-combined` on unstable).

The links should instead be created from a map that inserts the correct prefix for each platform. For the main ones, this would be:

- unstable
  - all except x86_64-linux and aarch64-linux: `nixpkgs/trunk`
- release (23.11 right now)
  - aarch64-darwin: `nixpkgs/nixpkgs-23.11-darwin` 
  - x86_64-darwin: `nixpkgs/nixpkgs-23.11-darwin` 

For other platforms there seems to be no dedicated hydra jobset right now.

Solving this issue might require searching for packages differently in the first place, which in turn could also fix #708

**Comment by vcunat** [comment]
And drop i686-linux.  For unstable, **all** links might be to `nixpkgs/trunk`, as it evaluates more often.

**Comment by vcunat** [comment]
Right.  You're not expected to run a 32-bit-only OS.  But we do support stuff like wine and steam (including whatever you need for them).  You'll end up with Hydra URLs that end with `.x86_64-linux` but the system really is `i686-linux`, e.g. https://hydra.nixos.org/job/nixpkgs/trunk/wine.x86_64-linux
